### PR TITLE
Sigs templating

### PR DESCRIPTION
**Pull request by xvzcf**
Since I needed to know how signatures would be listed in `generate.yml` for the interoperability tests, I decided to just finish all of the signature templating code.

I haven't added all the signatures yet. I'll do that along with the KEMS in one fell swoop. The separate commit adding `sphincs_haraka_128f_robust` was just done to test the templating mechanism.

### CODE REVIEW

**Comment by christianpaquin** [comment]
I'll take a look today

**Comment by xvzcf** [comment]
You also have to run `go run objects.go` in `crypto/obj` for the build to work. I'll add a wiki page with the instructions.

>  Perhaps you should rename `oqs_template` to `oqs-template` for consistency with openssl.

I used an underscore instead of a dash for internal consistency. [This top level folder for example](https://github.com/open-quantum-safe/boringssl/tree/master/third_party) uses underscores, whereas OpenSSL seems to prefer dashes for "compound" directory names.

**Comment by xvzcf** [comment]
Done.

File: README.md [hunk 1]
```diff
@@ -52,13 +52,13 @@ Some of the KEMs provided in liboqs do provide IND-CCA security; others do not (
...
 
-If an algorithm is provided by liboqs but is not listed below, it can still be used in the fork through [either one of two ways](https://github.com/open-quantum-safe/openssl/wiki/Using-liboqs-algorithms-that-are-not-in-the-forks).
+If an algorithm is provided by liboqs but is not listed below, it can still be used in the fork through [either one of two ways](https://github.com/open-quantum-safe/boringssl/wiki/Using-liboqs-algorithms-not-in-the-fork).
 
 #### Key Exchange
 
 The following quantum-safe algorithms from liboqs are supported (assuming they have been enabled in liboqs):
 
-- `oqs_kemdefault` (see [here](https://github.com/open-quantum-safe/openssl/wiki/Using-liboqs-algorithms-that-are-not-in-the-forks#oqsdefault) for what this denotes)
+- `oqs_kemdefault` (see [here](https://github.com/open-quantum-safe/boringssl/wiki/Using-liboqs-algorithms-not-in-the-fork#oqsdefault) for what this denotes)
```
Comment (christianpaquin) [inline README.md:61]: oqs_kem_default?

Comment (xvzcf) [reply README.md]: Yup, and it should be `oqs_sig_default` instead of `oqs_sigdefault`. Fixed.

**Pull request merged**